plugins/azure/virtualnetworks/noVPNgateways.js

```diff
@@ -0,0 +1,92 @@
+
+        async.each(locations.virtualNetworks, function(location, rcb){
+            var virtualNetworks = helpers.addSource(cache, source, 
+                ['virtualNetworks', 'listAll', location]);
+
+            if (!virtualNetworks) return rcb();
+
+            if (virtualNetworks.err || !virtualNetworks.data) {
+                helpers.addResult(results, 3, 'Unable to query for Virtual Networks: ' + helpers.addError(virtualNetworks), location);
+                return rcb();
+            }
+
+            if (!virtualNetworks.data.length) {
+                helpers.addResult(results, 0, 'No existing Virtual Networks found', location);
+                return rcb();
+            }
+            var resourceGroups = helpers.addSource(cache, source, 
+                ['resourceGroups', 'list', location]);
+
+            if (!resourceGroups || resourceGroups.err || !resourceGroups.data) {
+                helpers.addResult(results, 3, 'Unable to query for resource groups: ' + helpers.addError(resourceGroups), location);
+                return rcb();
+            }
+
+            if (!resourceGroups.data.length) {
+                helpers.addResult(results, 0, 'No existing resource groups found', location);
+                return rcb();
+            }
+
+            let vpnGatewaysList = [];
+            resourceGroups.data.forEach(resourceGroup => {
+                var virtualNetworkGateways = helpers.addSource(cache, source, 
+                    ['virtualNetworkGateways', 'listByResourceGroup', location, resourceGroup.id]);
+
+                if (!virtualNetworkGateways || virtualNetworkGateways.err || !virtualNetworkGateways.data) {
+                    helpers.addResult(results, 3, 'Unable to query for virtual Network Gateways: ' + helpers.addError(virtualNetworkGateways), location);
+                    return;
+                }
+                
+                if (virtualNetworkGateways.data.length) {
+
+                    for (let virtualNetworkGateway of virtualNetworkGateways.data) {
+                        if (virtualNetworkGateway.gatewayType && virtualNetworkGateway.gatewayType.toLowerCase()==='vpn')
+                            vpnGatewaysList.push({id:virtualNetworkGateway.id, name: virtualNetworkGateway.name });
+                    }
+                }
+            });
+            virtualNetworks.data.forEach(virtualNetwork => {
+                let vpnGatewayFound = [];
+                if (virtualNetwork.subnets.length) {
+                    for (let subnet of virtualNetwork.subnets) {
+
+                        if (subnet.properties && subnet.properties.ipConfigurations && subnet.properties.ipConfigurations.length) {
+                            for (let vpnGateway of vpnGatewaysList) {
+                                if (subnet.properties.ipConfigurations.some(configuration => (configuration.id.toLowerCase().indexOf(vpnGateway.id.toLowerCase()) > -1))) {
```

Comment:
```suggestion
                                if (subnet.properties.ipConfigurations.some(configuration => (configuration.id.toLowerCase().indexOf(vpnGateway.id.toLowerCase()) > -1))) {
```
should check if configuration.id and vpngateway.id exist before calling the tolowercase function